31-temperature-and-smoke-detector/firmware/src/main.cpp

```diff
@@ -0,0 +1,279 @@
+#include <Arduino.h>
+#include <Notecard.h>
+#include "SparkFunBME280.h"
+
+#include "STM32LowPower.h"
+
+#define APPLICATION_NAME        "nf31"
+#define DATA_FIELD_ALERT        "alert"
+#define DATA_FIELD_APP          "app"
+#define DATA_FIELD_TEMPERATURE  "temperature"
+#define DATA_FIELD_HUMIDITY     "humidity"
+#define DATA_FIELD_PRESSURE     "pressure"
+#define DATA_FIELD_GAS          "gas"
+#define APPLICATION_NOTEFILE    "mq2.qo"
+
+typedef struct {
+    double temperature;
+    double pressure;
+    double humidity;
+} BMESample;
+
+
+typedef struct MQ2SensorReading {
+    BMESample bme;
+    uint16_t gas;
+} MQ2SensorReading;
+
+static bool isAlert(MQ2SensorReading& reading);
+static bool registerNotefileTemplate();
+static void addNote(const MQ2SensorReading& reading, bool immediate, uint8_t alert);
+
+
+Notecard notecard;
+BME280 bmeSensor;
+// Uncomment this line and replace com.your-company:your-product-name with your
+// ProductUID.
+#define PRODUCT_UID "net.bowerham.kimball:temperaturesmokedetector"
```

Comment:
You left your productuid in here :)

Reply:
Well it was bound to happen eventually.  I had better delete that project!  Will push the fix too